prover/witness_generator/src/main.rs

```diff
@@ -76,6 +76,18 @@ async fn main() -> anyhow::Result<()> {
             .expect("Invalid Sentry URL")
             .with_sentry_environment(observability_config.sentry_environment);
     }
+    if let (Some(opentelemetry_level), Some(otlp_endpoint)) = (
```

Comment:
Honestly I am unsure whether OTLP configuration belongs to the observability config. [This](https://github.com/matter-labs/zksync-era/blob/f38c546d16c7ae192cfb9df1354a9364e3fdff3f/core/lib/env_config/src/observability.rs#L6) says that the logic is "historical" which makes me think this is the wrong place. That being said, I couldn't find where it would actually belong. Any ideas?

Reply:
In the past, observability configs were a part of `vlog` crate (e.g. it would implicitly load configs under the hood, having some implicit fallbacks), which was bad in many ways. The remark regarding historical logic is about this behavior -- we wanted to keep the logic to work exactly the same as it did before, because who knows where we can depend on such fallbacks.

The logic for configuration, however, now is in the right place -- next to other app configs.

I get the sentiment that this may appear to be boilerplate, but it is not. It's a valid way to instantiate observability stack *once per binary*. If we are to reduce boilerplate, we should think about reducing the amount of binaries (which is easily doable, has a lot of benefits, and should be encouraged) instead of saving a few lines here and there (which doesn't really have benefits besides saving a few lines that people will never look at after they're written because they just work, though at cost of most likely having worse interfaces or otherwise poor design decisions).